### PR TITLE
cubic: no rollback during slow start

### PR DESCRIPTION
**Pull request by junhochoi**
While the stored cwnd/ssthresh is in slow start state, don't try
rollback. This is not to delay to exit slow start when there is a loss.

### CODE REVIEW

**Review by LPardue — DISMISSED** [review]

File: src/recovery/cubic.rs [hunk 1]
```diff
@@ -385,6 +385,11 @@ fn checkpoint(r: &mut Recovery) {
 }
 
 fn rollback(r: &mut Recovery) -> bool {
+    // don't go back to slow start
```
Comment (ghedo) [inline src/recovery/cubic.rs:388]:
```suggestion
    // Don't go back to slow start.
```

File: src/recovery/cubic.rs [hunk 2]
```diff
@@ -928,7 +933,39 @@ mod tests {
             now + rtt + Duration::from_millis(5),
         );
 
-        // cwnd is restored to the previous one.
+        // This is from slow start, no rollback
```
Comment (ghedo) [inline src/recovery/cubic.rs:936]:
```suggestion
        // This is from slow start, no rollback.
```

File: src/recovery/cubic.rs [hunk 3]
```diff
@@ -928,7 +933,39 @@ mod tests {
...
+        assert_eq!(r.cwnd(), cur_cwnd);
+
+        let now = now + rtt;
+
+        // Trigger another congestion event
```
Comment (ghedo) [inline src/recovery/cubic.rs:941]:
```suggestion
        // Trigger another congestion event.
```

File: src/recovery/cubic.rs [hunk 4]
```diff
@@ -928,7 +933,39 @@ mod tests {
...
+        let rtt = Duration::from_millis(100);
+
+        let acked = vec![Acked {
+            pkt_num: 0,
+            // To exit from recovery
+            time_sent: now + rtt,
+            size: r.max_datagram_size,
+        }];
+
+        // Ack more than cwnd bytes with rtt=100ms
```
Comment (ghedo) [inline src/recovery/cubic.rs:958]:
```suggestion
        // Ack more than cwnd bytes with rtt=100ms.
```

File: src/recovery/cubic.rs [hunk 5]
```diff
@@ -928,7 +933,39 @@ mod tests {
...
+        r.update_rtt(rtt, Duration::from_millis(0), now);
+
+        // Trigger detecting sprurious congestion event
```
Comment (ghedo) [inline src/recovery/cubic.rs:961]:
```suggestion
        // Trigger detecting sprurious congestion event.
```

**Review by ghedo — DISMISSED** [review]
Added some nits but lgtm.

**Review by ghedo — DISMISSED** [review]

**Comment by junhochoi** [comment]
> Actually, I'm confused. The commit message says "don't rollback during slow start", but what the code does is avoiding rollback _to_ slow start. That is, the commit message seems to imply that the _current_ cwnd / ssthresh are used to determine whether to rollback or not, but the code uses the prior ones.
> 
> I think either the commit message or the code need to be updated to match each other.

Thanks. Current code is correct - I updated the commit message to match current behavior.

**Review by ghedo — APPROVED** [review]

**Pull request merged**